diff --git a/sam3d_objects/model/backbone/tdfy_dit/representations/mesh/flexicubes/flexicubes.py b/sam3d_objects/model/backbone/tdfy_dit/representations/mesh/flexicubes/flexicubes.py
index 583b5e3..8cc4a23 100644
--- a/sam3d_objects/model/backbone/tdfy_dit/representations/mesh/flexicubes/flexicubes.py
+++ b/sam3d_objects/model/backbone/tdfy_dit/representations/mesh/flexicubes/flexicubes.py
@@ -15,7 +15,6 @@
 # limitations under the License.
 import torch
 from .tables import *
-from kaolin.utils.testing import check_tensor
 
 __all__ = [
     'FlexiCubes'
@@ -56,6 +55,8 @@ class FlexiCubes:
 
     def __call__(self, voxelgrid_vertices, scalar_field, cube_idx, resolution, qef_reg_scale=1e-3,
                  weight_scale=0.99, beta=None, alpha=None, gamma_f=None, voxelgrid_colors=None, training=False):
+        from kaolin.utils.testing import check_tensor
+
         assert torch.is_tensor(voxelgrid_vertices) and \
             check_tensor(voxelgrid_vertices, (None, 3), throw=False), \
             "'voxelgrid_vertices' should be a tensor of shape (num_vertices, 3)"
diff --git a/sam3d_objects/pipeline/inference_pipeline.py b/sam3d_objects/pipeline/inference_pipeline.py
index d34d5c2..34f385b 100644
--- a/sam3d_objects/pipeline/inference_pipeline.py
+++ b/sam3d_objects/pipeline/inference_pipeline.py
@@ -382,6 +382,8 @@ class InferencePipeline:
     def init_slat_decoder_mesh(
         self, slat_decoder_mesh_config_path, slat_decoder_mesh_ckpt_path
     ):
+        if slat_decoder_mesh_config_path is None:
+            return None
         return self.instantiate_and_load_from_pretrained(
             OmegaConf.load(
                 os.path.join(self.workspace_dir, slat_decoder_mesh_config_path)
